### PR TITLE
Update Examples with the str.format() syntax.

### PR DESCRIPTION
**Pull request by SitiSchu**
conversationbot.py:
    Use `str.format()`

conversationbot2.py:
    Use `str.format()`
    make the states dict flake8 compatible.

echobot.py:
    Change Docstring to a Docstring using triple quotes.
    Add Docstrings to `main()` and `echo()`

echobot2.py:
    Add File Description to Docstring
    Add License notice to Docstring
    Add Docstrings to `start()`, `help()`, `echo()` and `main()`

inlinebot.py:
    Use `str.format()`
    Add File Description to Docstring
    Add License notice to Docstring
    Add Docstring to `start()`, `help()`, `inlinequery()`, `error()`
    Create the list "results" with the Inline Results instead of appending them.

inlinekeyboard.py:
    Use `str.format()`
    Put the Update in a `main()` function.

paymentbot.py:
    Use `str.format()`
    Shorten comment on how to get a provider token to avoid the comment being longer than 99 characters

timerbot.py:
    Use `str.format()`
    Add Docstrings to `alarm()`, `set_timer()`, `unset()`
    Rename `set()` to `set_timer()`  to avoud shadowing the built-in `set()`

### CODE REVIEW

File: examples/conversationbot.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def start(bot, update):
 
 def gender(bot, update):
     user = update.message.from_user
-    logger.info("Gender of %s: %s" % (user.first_name, update.message.text))
+    logger.info("Gender of {}: {}".format(user.first_name, update.message.text))
```
Comment (tsnoam) [inline examples/conversationbot.py]: Good catch here, we shouldn't be formatting logs ourselves but rather let logging do it. Unfortunately logging only work with `%s` notation. So please change to:
```
logger.info("Gender of %s: %s", user.first_name, update.message.text)
```
edit: I see that part of the logging occurrences you've done that way.

File: examples/conversationbot.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ def photo(bot, update):
     user = update.message.from_user
     photo_file = bot.get_file(update.message.photo[-1].file_id)
     photo_file.download('user_photo.jpg')
-    logger.info("Photo of %s: %s" % (user.first_name, 'user_photo.jpg'))
+    logger.info("Photo of {}: {}".format(user.first_name, 'user_photo.jpg'))
```
Comment (tsnoam) [inline examples/conversationbot.py]: same comment about logging

File: examples/conversationbot.py [hunk 3]
```diff
@@ -67,7 +67,7 @@ def photo(bot, update):
 
 def skip_photo(bot, update):
     user = update.message.from_user
-    logger.info("User %s did not send a photo." % user.first_name)
+    logger.info("User {} did not send a photo.".format(user.first_name))
```
Comment (tsnoam) [inline examples/conversationbot.py]: same comment about logging

File: examples/conversationbot.py [hunk 4]
```diff
@@ -77,8 +77,8 @@ def skip_photo(bot, update):
 def location(bot, update):
     user = update.message.from_user
     user_location = update.message.location
-    logger.info("Location of %s: %f / %f"
-                % (user.first_name, user_location.latitude, user_location.longitude))
+    logger.info("Location of {}: {} / {}".format(
+            user.first_name, user_location.latitude, user_location.longitude))
```
Comment (tsnoam) [inline examples/conversationbot.py]: same comment about logging

File: examples/conversationbot2.py [hunk 5]
```diff
@@ -41,24 +41,25 @@ def facts_to_str(user_data):
...
 
 
 def start(bot, update):
     update.message.reply_text(
-        "Hi! My name is Doctor Botter. I will hold a more complex conversation with you. "
-        "Why don't you tell me something about yourself?",
-        reply_markup=markup)
+            "Hi! My name is Doctor Botter. I will hold a more complex conversation with you. "
+            "Why don't you tell me something about yourself?",
+            reply_markup=markup)
```
Comment (tsnoam) [inline examples/conversationbot2.py]: why did you change the indentation? please revert.

File: examples/conversationbot.py [hunk 6]
```diff
@@ -87,7 +87,7 @@ def location(bot, update):
 
 def skip_location(bot, update):
     user = update.message.from_user
-    logger.info("User %s did not send a location." % user.first_name)
+    logger.info("User {} did not send a location.".format(user.first_name))
```
Comment (tsnoam) [inline examples/conversationbot.py]: same comment about logging

File: examples/conversationbot.py [hunk 7]
```diff
@@ -96,23 +96,24 @@ def skip_location(bot, update):
 
 def bio(bot, update):
     user = update.message.from_user
-    logger.info("Bio of %s: %s" % (user.first_name, update.message.text))
+    logger.info("Bio of {}: {}".format(user.first_name, update.message.text))
```
Comment (tsnoam) [inline examples/conversationbot.py]: same comment about logging

File: examples/conversationbot.py [hunk 8]
```diff
@@ -96,23 +96,24 @@ def skip_location(bot, update):
...
     update.message.reply_text('Thank you! I hope we can talk again some day.')
 
     return ConversationHandler.END
 
 
 def cancel(bot, update):
     user = update.message.from_user
-    logger.info("User %s canceled the conversation." % user.first_name)
+    logger.info("User {} canceled the conversation.".format(user.first_name))
```
Comment (tsnoam) [inline examples/conversationbot.py]: same comment about logging

File: examples/conversationbot2.py [hunk 9]
```diff
@@ -110,29 +111,29 @@ def main():
...
+                                ],
+
+                TYPING_REPLY: [MessageHandler(Filters.text,
+                                              received_information,
+                                              pass_user_data=True),
+                               ],
+                },
+
+            fallbacks=[RegexHandler('^Done$', done, pass_user_data=True)]
+            )
```
Comment (tsnoam) [inline examples/conversationbot2.py]: why did you change the indentation here? please revert

File: examples/inlinebot.py [hunk 10]
```diff
@@ -48,31 +50,29 @@ def escape_markdown(text):
...
+               InlineQueryResultArticle(id=uuid4(),
+                                        title="Bold",
+                                        input_message_content=InputTextMessageContent(
+                                                "*{}*".format(escape_markdown(query)),
+                                                parse_mode=ParseMode.MARKDOWN)),
+               InlineQueryResultArticle(id=uuid4(),
+                                        title="Italic",
+                                        input_message_content=InputTextMessageContent(
+                                                "_{}_".format(escape_markdown(query)),
+                                                parse_mode=ParseMode.MARKDOWN))]
```
Comment (tsnoam) [inline examples/inlinebot.py]: why did you change the indentation here? please revert.

File: examples/paymentbot.py [hunk 11]
```diff
@@ -31,7 +33,7 @@ def start_with_shipping_callback(bot, update):
     description = "Payment Example using python-telegram-bot"
     # select a payload just for you to recognize its the donation from your bot
     payload = "Custom-Payload"
-    # get your provider_token at @botfather, see https://core.telegram.org/bots/payments#getting-a-token
+    # to get your provider_token see https://core.telegram.org/bots/payments#getting-a-token
```
Comment (tsnoam) [inline examples/paymentbot.py]: If we're into grammar, "In order to get a provider_token ..." would be more appropriate.

File: examples/paymentbot.py [hunk 12]
```diff
@@ -55,7 +57,7 @@ def start_without_shipping_callback(bot, update):
     description = "Payment Example using python-telegram-bot"
     # select a payload just for you to recognize its the donation from your bot
     payload = "Custom-Payload"
-    # get your provider_token at @botfather, see https://core.telegram.org/bots/payments#getting-a-token
+    # check https://core.telegram.org/bots/payments#supported-currencies for more details
```
Comment (tsnoam) [inline examples/paymentbot.py]: Same comment about grammar.

File: examples/conversationbot2.py (hunk 9 above)
Comment (SitiSchu) [reply examples/conversationbot2.py]: I moved everything one space forward since a space between the double colons and the Square Brackets. If I remove that one space indent flake8 will complain.

File: examples/inlinebot.py (hunk 10 above)
Comment (SitiSchu) [reply examples/inlinebot.py]: I changed the entire code block here. Before my changes the Items were appended to the list, now the list is created with those items.

File: examples/inlinebot.py [hunk 13]
```diff
@@ -48,31 +50,29 @@ def escape_markdown(text):
...
+        InlineQueryResultArticle(id=uuid4(),
+                                 title="Bold",
+                                 input_message_content=InputTextMessageContent(
+                                    "*{}*".format(escape_markdown(query)),
+                                    parse_mode=ParseMode.MARKDOWN)),
+        InlineQueryResultArticle(id=uuid4(),
+                                 title="Italic",
+                                 input_message_content=InputTextMessageContent(
+                                     "_{}_".format(escape_markdown(query)),
+                                     parse_mode=ParseMode.MARKDOWN))]
```
Comment (tsnoam) [inline examples/inlinebot.py]: autopep8 can be of assistant here as well.

File: examples/paymentbot.py [hunk 14]
```diff
@@ -55,7 +57,8 @@ def start_without_shipping_callback(bot, update):
     description = "Payment Example using python-telegram-bot"
     # select a payload just for you to recognize its the donation from your bot
     payload = "Custom-Payload"
-    # get your provider_token at @botfather, see https://core.telegram.org/bots/payments#getting-a-token
+    # In order to get a provider_token to get your provider_token see
```
Comment (tsnoam) [inline examples/paymentbot.py]: In order to get a provider_token ~~to get your provider_token~~ see https://core.telegram.org/bots/payments#getting-a-token

**Comment by SitiSchu** [comment]
@tsnoam I made all the changes you requested,

File: examples/inlinebot.py (hunk 13 above)
Comment (SitiSchu) [reply examples/inlinebot.py]: autopep8 did this to that part:
ss 2017-10-15 at 10 49 27

This is how it looked before:
ss 2017-10-15 at 10 49 36

imho the way it was before is more readable

**Comment by SitiSchu** [comment]
I formated the InlineQuery results in inlinebot.py like @jh0ker suggested with a line before the `id=uuid4()`
This looks better and autopep8 didnt change the formatting to something weird.

File: examples/conversationbot.py [hunk 15]
```diff
@@ -96,23 +96,23 @@ def skip_location(bot, update):
...
+    logger.info("User %s canceled the conversation.", user.first_name)
     update.message.reply_text('Bye! I hope we can talk again some day.',
                               reply_markup=ReplyKeyboardRemove())
 
     return ConversationHandler.END
 
 
 def error(bot, update, error):
-    logger.warn('Update "%s" caused error "%s"' % (update, error))
+    logger.warn('Update "%s" caused error "%s"', update, error)
```
Comment (jh0ker) [inline examples/conversationbot.py]: Can you change this to `logger.warning`? `Logger.warn` is deprecated.
https://docs.python.org/3/library/logging.html#logging.Logger.warning

File: examples/conversationbot2.py [hunk 16]
```diff
@@ -90,15 +90,15 @@ def done(bot, update, user_data):
...
+                              "{}"
+                              "Until next time!".format(facts_to_str(user_data)))
 
     user_data.clear()
     return ConversationHandler.END
 
 
 def error(bot, update, error):
-    logger.warn('Update "%s" caused error "%s"' % (update, error))
+    logger.warn('Update "%s" caused error "%s"', update, error)
```
Comment (jh0ker) [inline examples/conversationbot2.py]: Can you change this to `logger.warning`? `Logger.warn` is deprecated.
https://docs.python.org/3/library/logging.html#logging.Logger.warning

File: examples/echobot2.py [hunk 17]
```diff
@@ -29,22 +29,27 @@
...
 
 
 def echo(bot, update):
+    """Echo the user message."""
     update.message.reply_text(update.message.text)
 
 
 def error(bot, update, error):
-    logger.warn('Update "%s" caused error "%s"' % (update, error))
+    logger.warn('Update "%s" caused error "%s"', update, error)
```
Comment (jh0ker) [inline examples/echobot2.py]: Can you change this to `logger.warning`? `Logger.warn` is deprecated.
https://docs.python.org/3/library/logging.html#logging.Logger.warning

File: examples/inlinebot.py [hunk 18]
```diff
@@ -48,31 +50,32 @@ def escape_markdown(text):
...
+            input_message_content=InputTextMessageContent(
+                "_{}_".format(escape_markdown(query)),
+                parse_mode=ParseMode.MARKDOWN))]
 
     update.inline_query.answer(results)
 
 
 def error(bot, update, error):
-    logger.warning('Update "%s" caused error "%s"' % (update, error))
+    logger.warn('Update "%s" caused error "%s"', update, error)
```
Comment (jh0ker) [inline examples/inlinebot.py]: Can you change this back to `logger.warning`? `Logger.warn` is deprecated.
https://docs.python.org/3/library/logging.html#logging.Logger.warning

File: examples/inlinekeyboard.py [hunk 19]
```diff
@@ -36,20 +36,26 @@ def help(bot, update):
 
 
 def error(bot, update, error):
-    logging.warning('Update "%s" caused error "%s"' % (update, error))
+    logger.warn('Update "%s" caused error "%s"', update, error)
```
Comment (jh0ker) [inline examples/inlinekeyboard.py]: Can you change this back to `logger.warning`? `Logger.warn` is deprecated.
https://docs.python.org/3/library/logging.html#logging.Logger.warning

File: examples/paymentbot.py [hunk 20]
```diff
@@ -10,13 +10,14 @@
...
-logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s', level=logging.INFO)
+logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+                    level=logging.INFO)
 
 logger = logging.getLogger(__name__)
 
 
 def error(bot, update, error):
-    logger.warn('Update "%s" caused error "%s"' % (update, error))
+    logger.warn('Update "%s" caused error "%s"', update, error)
```
Comment (jh0ker) [inline examples/paymentbot.py]: Can you change this to `logger.warning`? `Logger.warn` is deprecated.
https://docs.python.org/3/library/logging.html#logging.Logger.warning

File: examples/timerbot.py [hunk 21]
```diff
@@ -73,10 +73,11 @@ def unset(bot, update, chat_data):
 
 
 def error(bot, update, error):
-    logger.warning('Update "%s" caused error "%s"' % (update, error))
+    logger.warn('Update "%s" caused error "%s"', update, error)
```
Comment (jh0ker) [inline examples/timerbot.py]: Can you change this back to `logger.warning`? `Logger.warn` is deprecated.
https://docs.python.org/3/library/logging.html#logging.Logger.warning

**Comment by jh0ker** [comment]
@SitiSchu A bit more nitpicking :P

**Comment by SitiSchu** [comment]
@jh0ker PyCharm told me about this but I wasn't sure if Py2 supports it. I checked and it does support it so I'll change it today ^^.

**Comment by jh0ker** [comment]
@SitiSchu Awesome, thanks :)

**Comment by SitiSchu** [comment]
I updated all error functions to the new `warning` syntax. I also made flake8 pass(except `D103 Missing docstring in public function`)

File: examples/inlinebot.py [hunk 22]
```diff
@@ -44,7 +45,7 @@ def help(bot, update):
 
 
 def escape_markdown(text):
```
Comment (jh0ker) [inline examples/inlinebot.py]: Can we replace this as well? We have `telegram.utils.helpers.escape_markdown` now.

**Comment by SitiSchu** [comment]
@jh0ker Done.

**Comment by SitiSchu** [comment]
Any updates on this PR? :)

**Comment by Eldinnie** [comment]
LGTM @tsnoam any more changes requested?

**Comment by tsnoam** [comment]
@SitiSchu Thankyou for your contribution.
